Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR_test.py

```diff
@@ -439,3 +439,71 @@ def test_alert_process_search_command_func_called(mocker, args, func_to_be_calle
     execute_command = mocker.patch.object(CLIENT, func_to_be_called)
     process_search_command_with_polling(args, CLIENT)
     assert execute_command.called is True
+
+
+test_fetch_incidents_first_run_data = [
+    ({'results': [  # response
+        {'id': '123', 'backend_timestamp': '2023-05-12T08:16:51.779Z', 'first_event_timestamp': '2000-04-12T08:14:51.779Z'},
+        {'id': '456', 'backend_timestamp': '2023-05-12T08:17:51.779Z', 'first_event_timestamp': '2000-04-12T08:14:51.779Z'},
+        {'id': '789', 'backend_timestamp': '2023-05-12T08:17:51.779Z', 'first_event_timestamp': '2000-04-12T08:14:51.779Z'}
+    ]},
+        {'last_fetched_alert_create_time': '2023-05-12T08:17:51.779Z', 'last_fetched_alerts_ids': ['456', '789']}  # expected res
+    )
+]
+
+
+@pytest.mark.parametrize('response, expected_res', test_fetch_incidents_first_run_data)
+def test_fetch_incidents_first_run(mocker, response, expected_res):
+    """
+    Given:
+        - All arguments needed.
+
+    When:
+        - Running 'fetch-incidents' command in the first time.
+
+    Then:
+        - The fetch_incidents func returns the right last run.
+    """
+    from CarbonBlackEnterpriseEDR import fetch_incidents
+    mocker.patch.object(CLIENT, 'search_alerts_request', return_value=response)
+    mocker.patch('CommonServerPython.parse_date_range', return_value='2023-03-12T08:17:51.779Z')
+    # last_fetched_alert_create_time = '2024-07-12T08:08:17.000Z'
+    _, res = fetch_incidents(CLIENT, '3 days', '3', {})
+    assert res == expected_res
+
+
+test_fetch_incidents_second_run_data = [
+    ({'results': [  # response
+        {'id': '789', 'backend_timestamp': '2023-05-12T08:17:51.779Z', 'first_event_timestamp': '2000-04-12T08:14:51.779Z'},
+        {'id': '123', 'backend_timestamp': '2023-06-12T08:17:51.779Z', 'first_event_timestamp': '2000-05-12T08:14:51.779Z'},
+        {'id': '345', 'backend_timestamp': '2023-07-12T08:17:51.779Z', 'first_event_timestamp': '2000-05-12T08:14:51.779Z'}
+    ]},
+        {'last_fetched_alert_create_time': '2023-05-12T08:17:51.779Z', 'last_fetched_alerts_ids': ['456', '789']},  # last_run
+        [
+        {'name': 'Carbon Black Enterprise EDR alert 123', 'occurred': '2023-06-12T08:17:51.779Z'},
+        {'name': 'Carbon Black Enterprise EDR alert 345', 'occurred': '2023-07-12T08:17:51.779Z'},
+    ]
+    )
+]
+
+
+@pytest.mark.parametrize('response, last_run, expected_incidents', test_fetch_incidents_second_run_data)
+def test_fetch_incidents_second_run(mocker, response, last_run, expected_incidents):
+    """
+    Given:
+        - All arguments needed.
+
+    When:
+        - Running 'fetch-incidents' command in the second time (there is a last_run value).
+
+    Then:
+        - The fetch_incidents fun returns the right incidents.
```

Comment:
Please add an explanation: "should drop the duplicates and return only these".
please note the typo "fun"